Separate secret and private channels on whois for non-opers

B00mX0r opened the pull request:
This separates secret channels from non-secret channels when users WHOIS themselves.

This resolves #969.

SaberUK commented on src/coremods/core_whois.cpp:
```diff
@@ -177,10 +177,11 @@ void CommandWhois::SendChanList(WhoisContextImpl& whois)
+		 # a spy channel
 		 */
-		if ((whois.IsSelfWhois()) || ((!c->IsModeSet(privatemode)) && (!c->IsModeSet(secretmode))) || (c->HasUser(whois.GetSource())))
+		if ((!c->IsModeSet(privatemode) && !c->IsModeSet(secretmode)) || (c->HasUser(whois.GetSource()) && !whois.IsSelfWhois()))
```
It might also be desirable to split private/secret channels when the source is not the target user but is also on the channel. What do you think?

B00mX0r commented:
Since I couldn't see any harm and could only see benefits with making it so private and secret chans are also separated for other users whom you are in the same channel with, I went ahead and implemented that change.

Before merging, I'd also like to talk a bit about the RPL_CHANNELSMSG numeric to send the "is on private/secret channels" message. Where did this numeric come from? I cannot find any record of it anywhere else, and this is the only place in which it is used. My IRC client does not properly support it (unsurprising since it doesn't seem to originate from anywhere), and so the "is on private/secret channels" message is improperly separated from the rest of the WHOIS.

B00mX0r commented:
What if we do RPL_WHOISSPECIAL? I think that would make sense given that this is a special WHOIS line, and that this line is already supported by many clients.

SaberUK commented:
I've made some tweaks to this so that normal users can get split channel lists too. Can you see any issues with the changes i've made? I'm pretty sure they're okay but there might be some bugs I haven't spotted.

B00mX0r commented:
@SaberUK thanks for the changes. I changed the numeric to RPL_WHOISSPECIAL so clients would not break up the WHOIS reply, optimized an if loop, and removed an extra whitespace.

Other than that, looks good to me. I tested different config variations and oper/user scenarios.

SaberUK commented:
> I changed the numeric to RPL_WHOISSPECIAL so clients would not break up the WHOIS reply

This numeric was explicitly changed to an unused numeric to avoid collisions like this. Clients that don't currently support it can implement support for it if they want.

SaberUK commented on src/coremods/core_whois.cpp:
```diff
@@ -201,13 +216,13 @@ void CommandWhois::DoWhois(LocalUser* user, User* dest, time_t signon, unsigned
 	SendChanList(whois);
 
-	if (!whois.IsSelfWhois() && !ServerInstance->Config->HideServer.empty() && !user->HasPrivPermission("servers/auspex"))
+	if (whois.IsSelfWhois() || ServerInstance->Config->HideServer.empty() || user->HasPrivPermission("servers/auspex"))
```
This is unrelated to this PR so can you revert it for now?
